assets/js/base/components/cart-checkout/totals/coupon/index.tsx

```diff
@@ -63,30 +68,33 @@ export const TotalsCoupon = ( {
 		if ( currentIsLoading.current !== isLoading ) {
 			if ( ! isLoading && couponValue && ! validationError ) {
 				setCouponValue( '' );
+				setCouponFormClass( 'screen-reader-text' );
 			}
 			currentIsLoading.current = isLoading;
 		}
 	}, [ isLoading, couponValue, validationError ] );
 
 	return (
-		<Panel
-			className="wc-block-components-totals-coupon"
-			hasBorder={ false }
-			initialOpen={ initialOpen }
-			title={
-				<Label
-					label={ __(
-						'Coupon code',
+		<>
+			{ couponFormClass === 'screen-reader-text' && (
+				<a
+					role="button"
+					href="#wc-block-components-totals-coupon__form"
+					className="wc-block-components-totals-coupon-link"
+					aria-label={ __(
+						'Add a coupon',
 						'woo-gutenberg-products-block'
 					) }
-					screenReaderLabel={ __(
-						'Apply a coupon code',
-						'woo-gutenberg-products-block'
-					) }
-					htmlFor={ textInputId }
-				/>
-			}
-		>
+					onClick={ (
```

Comment:
This solution works but there is room for improvement and the opportunity to make it cleaner.

1. It's preferable to pass one function to the onClick prop, and handle the logic in that function.
```
onClick ={handleCouponAnchorClick}
```

```
handleCouponAnchorClick = (e:React.MouseEvent< HTMLAnchorElement, MouseEvent >) => {
e.preventDefault();
...
}

Reply:
Thank you for the suggestion. I have updated the code. 👍